keyvi/include/keyvi/dictionary/match.h

```diff
@@ -44,9 +45,9 @@ struct Match;
 namespace index {
 namespace internal {
 template <class MatcherT, class DeletedT>
-keyvi::dictionary::Match NextFilteredMatch(const MatcherT&, const DeletedT&);
+std::shared_ptr<keyvi::dictionary::Match> NextFilteredMatch(const MatcherT&, const DeletedT&);
```

Comment:
can this be already `match_t` if you move `using match_t = std::shared_ptr<Match>` after the forward declaration?